earthaccess/search.py

```diff
@@ -731,14 +732,31 @@ def point(self, lon: str, lat: str) -> Type[GranuleQuery]:
         super().point(lon, lat)
         return self
 
-    def polygon(self, coordinates: List[Tuple[str, str]]) -> Type[GranuleQuery]:
+    def polygon(
+        self,
+        coordinates: Union[
+            shapely.geometry.polygon.Polygon, str, bytes, List[Tuple[str, str]]
```

Comment:
```suggestion
            shapely.geometry.base.BaseGeometry, str, bytes, List[Tuple[str, str]]
```
I think since the rest of the implementation works of off the BaseGeometry, we could just use that here as well. We probably want to add some tests with different geometries and especially a clockwise vs couterclockwise ones

Reply:
Done (https://github.com/nsidc/earthaccess/pull/506/commits/583de40d931281bd149bf92afc81cdb5dca3347c). Will think about/work on tests later this week/next week.